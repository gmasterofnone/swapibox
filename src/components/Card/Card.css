.card {
  width: 350px;
  height: 530px;
  background-color: blue;
  border: 10px solid white;
  margin: 25px;
}

.image-container {
  display: flex;
  justify-content: flex-end;
}

.character-pic {
  background-color: purple;
  height: 330px;
  width: 350px;
  z-index: 998;
  position: relative;
}

.favorite-btn {
  background-color: transparent;
  position: absolute;
  height: 50px;
  width: 50px;
  z-index: 999;
}

.star {
  margin: 10px 15px 0 0;
}

.card-info {
  align-items: center;
  box-sizing: border-box;
  display: flex;
  justify-content: center;
  flex-direction: column;
  padding: 10px;
}

.character-name {
  text-align: center;
  margin: 5px 0 0 0;
  padding: none;
}

ul {
  list-style: none;
  /* margin: 0; */
  padding: 0;
}
li, h4 {
  color: white;
}

