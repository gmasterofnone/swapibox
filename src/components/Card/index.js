import React, { Component } from 'react';
import './Card.css'
import cardImage from './../Utils/cardImage.js';
import { filterCards } from '../Utils/API';
import PropTypes from 'prop-types';


class Card extends Component {
  constructor() {
    super()
    this.state = {
      favorite: false
    }
  }

  componentDidMount() {
    const { favorite } = this.props.cardData
    return favorite ? this.setState({ favorite }) : null
  }

  updateFavorites = (name, getFavorites) => {
    filterCards(name)
    this.setState( {favorite: !this.state.favorite} )
    getFavorites()
  }

  render() {
    const { favorite } = this.state;
    const { cardData, getFavorites } = this.props
    console.log(typeof getFavorites)
    const { name } = this.props.cardData;
    const description = Object.keys(cardData).map((spec, index) => {
      if (spec !== 'type' && spec !== 'name' && spec !== 'favorite' && !Array.isArray(cardData[spec])) {
        return <li key={Date.now() + index}>{spec}: {cardData[spec]}</li>
      } else if (Array.isArray(cardData[spec])) {
        return <li key={Date.now() + index}>{spec}: {cardData[spec].join(', ')}</li>
      }
      return '';
    })

    return(
      <div className='card'>
        <div className='image-container'>
          <div onClick={() => this.updateFavorites(name, getFavorites)}
          className='favorite-btn'
          >
            <img className='star' alt='star' src={`${cardImage[favorite]}`}/>
          </div>
          <img src={`${cardImage[name]}`} alt={name} className='character-pic'/>
        </div>
        <div className='card-info'>
          <h4 className='character-name'>{name.toUpperCase()}</h4>
          <ul>
            { description }
          </ul>
        </div>
      </div>
    )
  }
}

Card.propTypes = {
  cardData: PropTypes.object.isRequired
}

export default Card;