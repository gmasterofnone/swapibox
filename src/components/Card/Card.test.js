/* eslint-disable */

import React from 'react';
import { shallow } from 'enzyme';
import Card from '.';

describe('Card', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Card key={'Luke'} cardData={{name: 'Luke', homeworld: 'Tatooine', residents: ['Leia', 'Lando', 'Dave'], favorite: false }}/>);
  });


  it('matches the snapshot', () => { 
    expect(wrapper).toMatchSnapshot()
  });

  it('loads with the correct state', () => {
    const expected = {
      favorite: false
    }

    expect(wrapper.state()).toEqual(expected)
  });

  it('renders the correct number of lis', () => {
    expect(wrapper.find('ul').children().length).toEqual(2)
  });

});