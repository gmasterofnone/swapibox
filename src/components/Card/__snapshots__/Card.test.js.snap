// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    cardData={
      Object {
        "favorite": false,
        "homeworld": "Tatooine",
        "name": "Luke",
        "residents": Array [
          "Leia",
          "Lando",
          "Dave",
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="Luke"
          className="character-photo"
        />,
        <div
          className="card-info"
        >
          <ul>
            <li>
              name
              : 
              Luke
            </li>
            <li>
              homeworld
              : 
              Tatooine
            </li>
            <li>
              residents
              : 
              Leia, Lando, Dave
            </li>
            
          </ul>
        </div>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Luke",
          "className": "character-photo",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul>
            <li>
              name
              : 
              Luke
            </li>
            <li>
              homeworld
              : 
              Tatooine
            </li>
            <li>
              residents
              : 
              Leia, Lando, Dave
            </li>
            
          </ul>,
          "className": "card-info",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                name
                : 
                Luke
              </li>,
              <li>
                homeworld
                : 
                Tatooine
              </li>,
              <li>
                residents
                : 
                Leia, Lando, Dave
              </li>,
              "",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1539564172309",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "name",
                  ": ",
                  "Luke",
                ],
              },
              "ref": null,
              "rendered": Array [
                "name",
                ": ",
                "Luke",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "1539564172310",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "homeworld",
                  ": ",
                  "Tatooine",
                ],
              },
              "ref": null,
              "rendered": Array [
                "homeworld",
                ": ",
                "Tatooine",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "1539564172311",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "residents",
                  ": ",
                  "Leia, Lando, Dave",
                ],
              },
              "ref": null,
              "rendered": Array [
                "residents",
                ": ",
                "Leia, Lando, Dave",
              ],
              "type": "li",
            },
            "",
          ],
          "type": "ul",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Luke"
            className="character-photo"
          />,
          <div
            className="card-info"
          >
            <ul>
              <li>
                name
                : 
                Luke
              </li>
              <li>
                homeworld
                : 
                Tatooine
              </li>
              <li>
                residents
                : 
                Leia, Lando, Dave
              </li>
              
            </ul>
          </div>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Luke",
            "className": "character-photo",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul>
              <li>
                name
                : 
                Luke
              </li>
              <li>
                homeworld
                : 
                Tatooine
              </li>
              <li>
                residents
                : 
                Leia, Lando, Dave
              </li>
              
            </ul>,
            "className": "card-info",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  name
                  : 
                  Luke
                </li>,
                <li>
                  homeworld
                  : 
                  Tatooine
                </li>,
                <li>
                  residents
                  : 
                  Leia, Lando, Dave
                </li>,
                "",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "1539564172309",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "name",
                    ": ",
                    "Luke",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "name",
                  ": ",
                  "Luke",
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "1539564172310",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "homeworld",
                    ": ",
                    "Tatooine",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "homeworld",
                  ": ",
                  "Tatooine",
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "1539564172311",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "residents",
                    ": ",
                    "Leia, Lando, Dave",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "residents",
                  ": ",
                  "Leia, Lando, Dave",
                ],
                "type": "li",
              },
              "",
            ],
            "type": "ul",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
